architecture.md

```diff
@@ -0,0 +1,78 @@
+# Ignition Gazebo Architecture
+
+Ignition Gazebo is the application entry-point of Ignition architecture, generally encompassing the use of all other Ignition libraries.
+It is a library itself, but its use case described in this document is as an executable.
+As an executable, it runs a simulation by launching two processes: the backend server process and frontend client process.
+
+The core of Ignition Gazebo is the server-client communication and the various plugins it loads to facilitate the simulation.
+Plugins are loaded at runtime to create a link between other Ignition libraries and Ignition Gazebo so they can exchange information.
+Users can write their own plugins.
+Because they’re loaded at runtime, Ignition does not need to be recompiled to add or remove plugins.
+There are other types of plugins in the scope of Ignition; however, this use case only concerns Ignition Gazebo server and GUI plugins.
```

Comment:
Is "this use case" referring to the "other types of plugins," or the type described earlier in the paragraph?

Reply:
Hmm.. I guess what I meant was "In the scope of this article", referring back to line 4 where I said "its [Ignition Gazebo's] use case described in this document is as an executable." 

Or possibly "the use case of Ignition Gazebo as an executable" makes more sense for line 11. I'm not sure if either of those re-phrasings is accurate enough to state.